lms/djangoapps/discussion/views.py

```diff
@@ -153,10 +153,9 @@ def get_threads(request, course, user_info, discussion_id=None, per_page=THREADS
     # If not provided with a discussion id, filter threads by commentable ids
     # which are accessible to the current user.
     if discussion_id is None:
-        discussion_category_ids = set(utils.get_discussion_categories_ids(course, request.user))
```

Comment:
Would previously call [`this`](https://github.com/edx/edx-platform/blob/335696a188ee6eee746392f978d59c66f04534b4/lms/djangoapps/django_comment_client/utils.py#L442), which went into [`this`](https://github.com/edx/edx-platform/blob/335696a188ee6eee746392f978d59c66f04534b4/lms/djangoapps/django_comment_client/utils.py#L119), which was loading *all* discussion xblocks for a course in order to check access.